pdns/zonemd.cc

```diff
@@ -65,26 +23,140 @@ void pdns::zonemdVerify(const DNSName& zone, ZoneParserTNG& zpt, bool& validatio
       std::string err = "Bad record content in record for '" + dnsResourceRecord.qname.toStringNoDot() + "|" + dnsResourceRecord.qtype.toString() + "': " + e.what();
       throw PDNSException(err);
     }
-    if (dnsResourceRecord.qtype == QType::SOA && dnsResourceRecord.qname == zone) {
-      soaRecordContent = std::dynamic_pointer_cast<SOARecordContent>(drc);
+    DNSRecord rec;
+    rec.d_name = dnsResourceRecord.qname;
+    rec.d_content = drc;
+    rec.d_type = dnsResourceRecord.qtype;
+    rec.d_class = dnsResourceRecord.qclass;
+    rec.d_ttl = dnsResourceRecord.ttl;
+    rec.d_clen = dnsResourceRecord.content.length(); // XXX is this correct?
+    readRecord(rec);
```

Comment:
I imagined it the other way around, passing the name, type and content instead but this works as well :)